comp-file/src/main/java/org/jumpmind/metl/core/runtime/component/S3ObjectWriter.java

```diff
@@ -0,0 +1,93 @@
+package org.jumpmind.metl.core.runtime.component;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+
+import org.jumpmind.metl.core.runtime.Message;
+import org.jumpmind.metl.core.runtime.flow.ISendMessageCallback;
+import org.jumpmind.metl.core.runtime.resource.IDirectory;
+import org.jumpmind.metl.core.runtime.resource.aws.S3BucketOperations;
+
+import software.amazon.awssdk.services.s3.model.PutObjectResponse;
+
+public class S3ObjectWriter extends AbstractFileWriter {
+    public static final String HEADER_SOURCE_FILE = "S3ObjectSourceFileName";
+
+    private S3BucketOperations bucket;
+
+    @Override
+    public void start() {
+        init();
+
+        bucket = (S3BucketOperations) getResourceReference();
+        if (bucket == null) {
+            throw new IllegalStateException("S3 resource has not been configured.");
+        }
+    }
+
+    @Override
+    public boolean supportsStartupMessages() {
+        return false;
+    }
+
+    @Override
+    public void handle(final Message inputMessage, final ISendMessageCallback callback,
+            final boolean unitOfWorkBoundaryReached) {
+        String sourceFileName = inputMessage.getHeader().getAsStrings().get(HEADER_SOURCE_FILE);
+        if (sourceFileName != null) {
+            File file = new File(sourceFileName);
+            String objectKey = getFileName(inputMessage);
```

Comment:
Does this need to be guarded by a check for `inputMessage instanceof ContentMessage<?>` in order to work as expected?

Reply:
yes